Resources/Prototypes/_NF/Entities/Objects/Weapons/Guns/turrets.yml

```
@@ -0,0 +1,313 @@
+   A Self-Recharging Laser Auto Turret (SLAT). A laser turret with built-in microreactor. Use screwdriver to disassemble.
+  suffix: Frontier, NT
+  components:
+  - type: Sprite
+    sprite: _NF/Objects/Weapons/Guns/Turrets/laser.rsi
+  - type: Gun
+    fireRate: 2
+    soundGunshot:
+      path: /Audio/Weapons/Guns/Gunshots/laser.ogg
+      params:
+        variation: 0.125
+  - type: HitscanBatteryAmmoProvider
+    proto: RedLightLaser
+    fireCost: 10
+  - type: NpcFactionMember
+    factions:
+    - NanoTrasen
+  - type: Construction
+    deconstructionTarget: null
+    graph: WeaponTurretLaserNanoTrasenPackingGraph
+    node: WeaponTurretLaserNanoTrasenPacked
+
+- type: entity # folded turret
+  parent: BaseItem
+  id: WeaponTurretLaserNanoTrasenPacked
+  name: packed SLAT
+  description: |-
+   A disassembled and packed Self-Recharging Laser Auto Turret (SLAT).
+  suffix: NT, Packed
+  components:
+  - type: WeaponCaseInsertable
+  - type: MultiHandedItem
+  - type: SpawnAfterInteract
+    prototype: WeaponTurretLaserNanoTrasenDeployed
+    doAfter: 5
+    removeOnInteract: true
+  - type: Sprite
+    sprite: _NF/Objects/Weapons/Guns/Turrets/laser.rsi
+    state: packed
+  - type: Item
+    size: Ginormous
+    shape:
+    - 0,0,3,3
+    sprite: _NF/Objects/Weapons/Guns/Turrets/laser.rsi
+
+- type: constructionGraph # deconstruction graph
+  id: WeaponTurretLaserNanoTrasenPackingGraph
+  start: WeaponTurretLaserNanoTrasenPacked
+  graph:
+  - node: WeaponTurretLaserNanoTrasenPacked
+    entity: WeaponTurretLaserNanoTrasenDeployed
+    edges:
+    - to: WeaponTurretLaserNanoTrasenPacked
+      completed:
+      - !type:SpawnPrototype
+            prototype: WeaponTurretLaserNanoTrasenPacked
+            amount: 1
+      - !type:EmptyAllContainers
+      - !type:DeleteEntity {}
+      steps:
+        - tool: Screwing
+          doAfter: 14
+
+- type: entity
+  parent: BaseWeaponTurretEnergyNF
+  id: WeaponTurretLaserSyndicateNF
+  name: laser turret
+  suffix: Frontier, Syndicate
+  components:
+  - type: Sprite
+    sprite: _NF/Objects/Weapons/Guns/Turrets/laser.rsi
+  - type: Gun
+    fireRate: 2
+    soundGunshot:
+      path: /Audio/Weapons/Guns/Gunshots/laser.ogg
+      params:
+        variation: 0.125
+  - type: HitscanBatteryAmmoProvider
+    proto: RedLightLaser
+    fireCost: 10
+  - type: NpcFactionMember
+    factions:
+    - Syndicate
+
+- type: entity
+  parent: WeaponTurretLaserSyndicateNF
+  id: WeaponTurretLaserHostileNF
+  name: laser turret
+  suffix: Frontier, Hostile
+  components:
+  - type: NpcFactionMember
+    factions:
+    - SimpleHostile
+
+## Magazine fed, deployable NanoTrasen turret
+- type: entity # deployed turret
+  parent:
+  - BaseWeaponTurretMagazineFed
+  id: WeaponTurretAsmgtNtDeployed
+  name: ASMGT
+  description: |-
+   An Autonomous Sentry Machine Gun Turret (ASMGT) is a magazine fed ballistic turret compatible with .20, .25, .30 and .35 cartriges. Use screwdriver to disassemble.
```

I am once again requesting plausibility. :pray:
```suggestion
   An Autonomous Sentry Machine Gun Turret (ASMGT) is a magazine-fed turret with barrels compatible with .20, .25, .30 and .35 cartriges. Use screwdriver to disassemble.
```

---

lissen, this is far future, no explanation is needed. jk.

I think something like this can be achieved with a single barrel, but any cartridges of smaller caliber need to be clamped in some plastic to fit in the chamber snuggly. Can pretend athe such system is part of that box magazine.

---

But lets go with your version.